src/internal/transactionenv/txncontext/context.go

```diff
@@ -5,17 +5,18 @@ import (
 
 	"github.com/gogo/protobuf/types"
 	"github.com/jmoiron/sqlx"
-
+	"github.com/pachyderm/pachyderm/v2/src/auth"
+	"github.com/pachyderm/pachyderm/v2/src/internal/uuid"
 	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"github.com/pachyderm/pachyderm/v2/src/pps"
 )
 
 // TransactionContext is a helper type to encapsulate the state for a given
 // set of operations being performed in the Pachyderm API.  When a new
 // transaction is started, a context will be created for it containing these
 // objects, which will be threaded through to every API call:
 type TransactionContext struct {
-	// ClientContext is the incoming context.Context for the request.
-	ClientContext context.Context
+	username string
```

Comment:
Why not export this?  This is great, we should use this when we are in a transaction.

Reply:
I didn't want to make it easy to modify, but maybe that's not an actual concern. The `WhoAmI` method I exposed seemed like a simple dropin, anyway.

Reply:
Okay, makes sense to me.